arch/arm/dts/imx6-apalis.dts

```diff
@@ -14,7 +14,7 @@
 	/* Will be filled by the bootloader */
 	memory@10000000 {
 		device_type = "memory";
-		reg = <0x10000000 0>;
+		reg = <0x10000000 0x40000000>;
```

Comment:
I'm trying to understand why this was set to 0 initially, and looking at the upstream kernel that still seems to be the case (and probably why u-boot is reflecting the same value).

Did you have to change so OP-TEE would end up having the right amount? How would this work with modules that have different memory layouts?

I wonder if the right fix (might be hard due size constrains) is for SPL to update and populate with the correct size as identified during runtime (based on the module type/revision).

Reply:
> How would this work with modules that have different memory layouts?
yes, this is what I missed (there are different version of Apalis with different amount of RAM)

>  populate with the correct size as identified during runtime (based on the module type/revision).
yes, let me investigate how to do that

Reply:
Looks like this change was not needed (maybe I somehow messed up with config options, so and changed the logic of providing mem information for OP-TEE). I've just compiled everything from scratch, also disabled CAAM and OP-TEE just boots without any issues.